perfkitbenchmarker/providers/ibmcloud/utils.py

```diff
@@ -0,0 +1,445 @@
+
+  def __init__(self, resource, *args):
+    """Initializes a IbmAPICommand with the provided args and common flags.
+
+    Args:
+      resource: A Genesis resource of type BaseResource.
+      *args: sequence of strings. Positional args to pass to genesis cli,
+      typically specifying an operation to perform (e.g. ['image', 'list']
+      to list available images).
+    """
+    if args and args[0] == 'login':
+      IbmAPICommand.gen = ibmcloud.IbmCloud(account=self.ibmcloud_account_id,
+                                            apikey=self.ibmcloud_apikey,
+                                            verbose=self.debug,
+                                            version='v1',
+                                            silent=True,
+                                            force=False)
+      IbmAPICommand.gen_instmgr = ibmcloud.InstanceManager(IbmAPICommand.gen)
+      IbmAPICommand.gen_imgmgr = ibmcloud.ImageManager(IbmAPICommand.gen)
+      IbmAPICommand.gen_fipmgr = ibmcloud.FipManager(IbmAPICommand.gen)
+      IbmAPICommand.gen_keymgr = ibmcloud.KeyManager(IbmAPICommand.gen)
+      IbmAPICommand.gen_subnetmgr = ibmcloud.SubnetManager(IbmAPICommand.gen)
+      IbmAPICommand.gen_volumemgr = ibmcloud.VolumeManager(IbmAPICommand.gen)
+      IbmAPICommand.gen_vpcmgr = ibmcloud.VPCManager(IbmAPICommand.gen)
+      IbmAPICommand.gen_sgmgr = ibmcloud.SGManager(IbmAPICommand.gen)
+    self.args = list(args)
+    self.flags = {}
+    self.user_data = None
+    self.sgid = None
+
+  def GetToken(self):
+    """Returns user token """
+    return self.gen.GetToken()
+
+  def CreateSgRules(self):
+    """Creates some default security group rules """
+    sgid = self.flags['sgid']
+    self.gen_sgmgr.CreateRule(sgid, 'inbound', 'ipv4', '0.0.0.0/0', 'icmp', None)
+    self.gen_sgmgr.CreateRule(sgid, 'inbound', 'ipv4', '0.0.0.0/0', 'tcp', 22)
+    self.gen_sgmgr.CreateRule(sgid, 'inbound', 'ipv4', '0.0.0.0/0', 'tcp', None,
+                              port_min=1024, port_max=50000)
+    self.gen_sgmgr.CreateRule(sgid, 'inbound', 'ipv4', '0.0.0.0/0', 'udp', None,
+                              port_min=1024, port_max=50000)
+    self.gen_sgmgr.CreateRule(sgid, 'inbound', 'ipv4', '0.0.0.0/0', 'tcp', 443)
+
+  def GetSgId(self):
+    """returns sg id """
+    logging.info('Looking up existing default sg')
+    prefix = self.flags['prefix']
+    resp = self.gen_sgmgr.List()
+    if resp and 'security_groups' in resp:
+      for item in resp['security_groups']:
+        if item['vpc']['name'].startswith(prefix):  # find the one with matching vpc name
+          return item['id']
+    else:
+      logging.error('Failed to retrieve security group id: %s', resp)
+    return None
+
+  def CreatePrefix(self):
+    """Creates address prefix on vpc """
+    return self.gen_vpcmgr.CreatePrefix(self.flags['vpcid'], self.flags['zone'],
+                                        self.flags['cidr'], name=self.flags['name'])
+
+  def CreateSubnet(self):
```

Comment:
Should this be in IBM network?

Reply:
yes, this is needed in ibm cloud network, this is created on a vpc where subnets are attached, address prefix must be added to vpc before subnets can be attached to vpc.